UCL-INGI/INGInious
# Refactor code for proper packaging

I made an attempt at packaging INGInious with setuptools.
The code can now be installed using pip.

To try it, you better use a virtualenv :-)
```
mkvirtualenv -p /usr/bin/python2 ingitest
workon ingitest

cd INGInious_source
pip install .

cd some_inginious_root_outside_of source
app_frontend.py # will read configuration.json in $(pwd)
                # and get the task folder location from it.
```

I have several remaining problems :

 1. I could not run the tests in my branch (nor in master). This is required to ensure that my modifications did not break anything.
 2. Integration with webserver and such could be greatly improved. 
 3. The test suite could be standardized and integrated into setup.py
 4. ... :-)

Main advantage : easy deployment with dependency management.
The installation instructions are greatly reduced : 

 - install mongodb & docker; enable both.
 - install INGInious (`pip install INGInious` when the package will be released)
 - configure configuration.json
 - (optionnal) configure lighthttpd + fastcgi.
